my-code/src/Contexts/MovieDetails.tsx

```diff
@@ -0,0 +1,55 @@
+import axios from 'axios';
+import { createContext, ReactNode, useContext, useState } from 'react';
+
+interface MovieDetailsProviderProps {
+    children: ReactNode
+}
+
+interface MovieDetailsContextProps {
+    getMovieInfo: (imdbID: string) => Promise<void>
+    movieInfo: MovieInfoProps
+    isLoading: boolean
+}
+
+interface MovieInfoProps {
+    imdbID: string
+    Runtime: string
+    Year: string
+    Rated: string
+    Title: string
+    Ratings: [{ Source: string, Value: string }]
+    Plot: string
+    Actors: string;
+    Genre: string;
+    Director: string
+    Poster: string
+}
+
+const MovieDetailsContext = createContext({} as MovieDetailsContextProps)
```

Comment:
This can cause errors, e.g. someone trying to call getMovieInfo, and receiving an error because undefined is not callable.

```suggestion
const MovieDetailsContext = createContext<MovieDetailsContextProps>({
  getMovieInfo: async () => {
    return;
  },
  movieInfo: {},
  isLoading: false,
});
```

Reply:
`movieInfo` is empty in this case, resulting in a type error. You could make all `MovieInfoProps` values be potentially undefined, or you could say that `movieInfo` can be undefined/null. Your call!